spark3-extensions/src/main/scala/org/apache/spark/sql/catalyst/utils/RewriteRowLevelOperationHelper.scala

```diff
@@ -229,4 +232,26 @@ object RewriteRowLevelOperationHelper {
   private final val AFFECTED_FILES_ACC_NAME = "internal.metrics.merge.affectedFiles"
   private final val AFFECTED_FILES_ACC_ALIAS_NAME = "_affectedFiles_"
   private final val SUM_ROW_ID_ALIAS_NAME = "_sum_"
+
+  def createScanRelation(
+      relation: DataSourceV2Relation,
+      scan: Scan,
+      outputAttrs: Seq[AttributeReference]): DataSourceV2ScanRelation = {
+    val ctor: DynConstructors.Ctor[DataSourceV2ScanRelation] =
```

Comment:
This constructor should be moved into a companion object so that the lookups don't happen every time the method is called.

Reply:
The `createScanRelation` method is already in the companion object. I moved the val defining the constructor to the object too, so it is not created each time the method is called.